www/src/pages/editor/NewMineralSiteModal.tsx

```diff
@@ -0,0 +1,384 @@
+import { GradeTonnage } from "../../../models/mineralSite/GradeTonnage";
+import { CommodityStore } from "models/commodity";
+import { DedupMineralSite, DedupMineralSiteLocation } from "../../../models/dedupMineralSite";
+import axios from "axios";
+import { Country, StateOrProvince } from "models";
+import { useEffect, useState } from "react";
+import { v4 as uuidv4 } from "uuid";
+import { UserStore } from "models/user";
+
+interface NewMineralSiteModalProps {
+  commodity: Commodity;
+  visible: boolean;
+  onClose: () => void;
+}
+
+export const NewMineralSiteModal: React.FC<NewMineralSiteModalProps> = ({ commodity, visible, onClose }) => {
+  const { mineralSiteStore, dedupMineralSiteStore, userStore, commodityStore, countryStore, stateOrProvinceStore, depositTypeStore, unitStore } = useStores();
+  const [form] = Form.useForm();
+  const [commodityOptions, setCommodityOptions] = useState<{ value: string; label: string }[]>([]);
+  const [countryOptions, setCountryOptions] = useState<{ value: string; label: string }[]>([]);
+  const [stateOptions, setStateOptions] = useState<{ value: string; label: string }[]>([]);
+  const [depositTypeOptions, setDepositTypeOptions] = useState<{ value: string; label: string }[]>([]);
+  const [unitOptions, setUnitOptions] = useState<{ value: string; label: string }[]>([]);
+  const [recordId, setRecordId] = useState("");
+  const [selectedSourceType, setSelectedSourceType] = useState<string | null>(null);
+
+  useEffect(() => {
+    commodityStore.fetchCriticalCommotities().then((commodities) => {
+      setCommodityOptions(commodities.map((comm: { id: string; name: string }) => ({ value: comm.id, label: comm.name })));
+    });
+
+    countryStore.fetchAll().then(() => {
+      setCountryOptions(
+        Array.from(countryStore.records.values())
+          .filter((country): country is Country => country !== null)
+          .map((country: Country) => ({ value: country.id, label: country.name }))
+      );
+    });
+
+    stateOrProvinceStore.fetchAll().then(() => {
+      setStateOptions(
+        Array.from(stateOrProvinceStore.records.values())
+          .filter((state): state is StateOrProvince => state !== null)
+          .map((state: StateOrProvince) => ({ value: state.id, label: state.name }))
+      );
+    });
+
+    depositTypeStore.fetchAll().then(() => {
+      setDepositTypeOptions(
+        Array.from(depositTypeStore.records.values())
+          .filter((type): type is DepositType => type !== null)
+          .map((type) => ({ value: type.id ?? "", label: type.name ?? "Unnamed" }))
+      );
+    });
+    unitStore.fetchAll().then(() => {
+      setUnitOptions(
+        Array.from(unitStore.records.values())
+          .filter((unit): unit is Unit => unit !== null)
+          .map((unit) => ({ value: unit.id ?? "", label: unit.name ?? "Unnamed" }))
+      );
+    });
+    const generatedId = `record-${uuidv4()}`;
```

Comment:
will it create a bug after you create one mineral site, you click to create another mineral site, if this component is not dropped, we have same record id